File: specification/specs2/cluster/cluster_settings/cluster_put_settings/ClusterPutSettingsRequest.ts
@rest_spec_name("cluster.put_settings")
class ClusterPutSettingsRequest extends RequestBase {
	persistent: Map<string, any>;
	transient: Map<string, any>;
	@request_parameter()
	flat_settings: boolean;
	@request_parameter()
	master_timeout: Time;
	@request_parameter()
	timeout: Time;
}

File: output/csharp/Cluster/ClusterSettings/ClusterPutSettings/ClusterPutSettingsRequest.cs
using Nest.CommonOptions;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.Cluster {

	public class ClusterPutSettingsRequest : RequestBase {
		
		[DataMember(Name="flat_settings")]
		public bool FlatSettings { get; set; }


		[DataMember(Name="master_timeout")]
		public Time MasterTimeout { get; set; }


		[DataMember(Name="timeout")]
		public Time Timeout { get; set; }


		[DataMember(Name="persistent")]
		public IDictionary<string, object> Persistent { get; set; }


		[DataMember(Name="transient")]
		public IDictionary<string, object> Transient { get; set; }

	}
}
